jupyterhub/spawner.py

```diff
@@ -424,6 +425,13 @@ def _options_from_form(self):
     def _default_options_from_form(self, form_data):
         return form_data
 
+    def run_options_from_form(self, form_data):
+        sig = signature(self.options_from_form)
+        if 'spawner' in sig.parameters:
+            return self.options_from_form(form_data, spawner=self)
+        else:
+            return self.options_from_form(form_data)
```

Comment:
Perfect! Can you add a test to test_spawner.py that exercises both forms of `options_from_form`? I was a little surprised the previous form of this PR didn't make anything fail, which means we didn't have coverage.

Sample form data is always a dict of lists of strings:

```python
from_data = {"key": ["value"]}
```

Reply:
I added 2 tests, one for a case with a spawner argument and one without.  Does that look OK?  Happy to improve it if it needs to do more.